### DISCUSSION TITLE
Tri-state palette toggle for system-preference or user-preference

### DISCUSSION THREAD

**Discussion by phaseOne** [squidfunk/mkdocs-material#3878] [Ideas]
It's awesome that this project has [implemented an option](https://squidfunk.github.io/mkdocs-material/setup/changing-the-colors/#color-palette-toggle) to change the palette by a toggle, use media queries, or both.

Though after you click the toggle once, it's a little clunky if you want to clear the state since it's stored in LocalStorage. I think upgrading the toggle to a dropdown would provide an even better user experience. This way, the user could choose between automatic switching by their OS or instead override with their preferred style _and_ be able to switch back. I think a good example is the toggle in the MDN docs:

<img width="180" alt="CleanShot 2022-05-01 at 22 12 30@2x" src="https://user-images.githubusercontent.com/1310985/166192364-dd063296-873a-4088-ba7b-4efac9a8a066.png">

_x-link https://github.com/squidfunk/mkdocs-material/discussions/2420_

**Comment by squidfunk** [comment]
Thanks for suggesting. This was already discussed several times:

- #3651
- #2727
- #3123
- #3244
